pkgs/by-name/md/mdwatch/package.nix

```
@@ -0,0 +1,33 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  nix-update-script,
+  testers,
+  mdwatch,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "mdwatch";
+  version = "0.1.11";
+
+  src = fetchFromGitHub {
+    owner = "santoshxshrestha";
+    repo = "mdwatch";
+    rev = "v${version}";
+    hash = "sha256-v+QsSPknUAFk+GC8D0wtA8VTcCWeb8DX6inL6WNl8gQ=";
+  };
+
+  cargoHash = "sha256-4Xvi5DxlqxfwnIa00Dr7tRzYa8/52BH/SODiNenlFDg=";
+
+  updateScript = nix-update-script { };
+
+  passthru.tests.version = testers.testVersion { package = mdwatch; };
```

versionCheckHook is preferred because it causes the build to fail if it fails whereas testVersion failing often gets missed. 
```suggestion
  doInstallCheck = true;
  nativeInstallCheckInputs = [ versionCheckHook ];
```